runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java

```diff
@@ -1132,8 +1132,14 @@ public synchronized CompletableFuture<Layout> invalidateLayout() {
             List<String> servers = Optional.ofNullable(latestLayout)
                     .map(Layout::getLayoutServers)
                     .orElse(bootstrapLayoutServers);
-
-            layout = fetchLayout(servers);
+            final List<String> responsiveServer =
+                    servers.stream().filter(server -> !latestLayout.getUnresponsiveServers()
+                            .contains(server))
+                            .collect(Collectors.toList());
+            if (responsiveServer.isEmpty()) {
+                throw new IllegalStateException("All servers are unresponsive");
```

Comment:
Just curious if you have checked all usages of `invalidateLayout` and make sure they handle the exception properly?

Reply:
@Lujie1996 
We should never get this state, hence it's an `IllegalStateException`, all the calls to it handle it as a generic exception exception for one place which I fixed.